client/pkg/eventlog/eventlog_impl.go

```diff
@@ -385,6 +385,45 @@ func (w *logWriter) doAppend(ctx context.Context, event *ce.Event) (int64, error
 	return offset, nil
 }
 
+func (w *logWriter) SyncAppendStream(ctx context.Context, event *ce.Event) (int64, error) {
+	// TODO: async for throughput
+
+	retryTimes := defaultRetryTimes
+	for i := 1; i <= retryTimes; i++ {
+		offset, err := w.doSyncAppendStream(ctx, event)
+		if err == nil {
+			return offset, nil
+		}
+		vlog.Warning(ctx, "failed to Append", map[string]interface{}{
+			vlog.KeyError: err,
+			"offset":      offset,
+		})
+		if errors.Is(err, errors.ErrSegmentFull) {
+			if i < retryTimes {
+				continue
+			}
+		}
+		return -1, err
+	}
+
+	return -1, errors.ErrUnknown
```

Comment:
Don't return Unknown, use a specific one.

Reply:
Unless the defaultRetryTimes is 0, it will not enter this branch. At present, it can only be defined as an unknown error. After the segment route handoff is merged, this will be changed